src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SafeEvpPKeyHandle.OpenSsl.macOS.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.Versioning;
+using System.Threading;
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class SafeEvpPKeyHandle
+    {
+        private static long _openSslVersion;
+
+        /// <summary>
+        /// The runtime version number for the loaded version of OpenSSL.
+        /// </summary>
+        /// <remarks>
+        /// For OpenSSL 1.1+ this is the result of <code>OpenSSL_version_num()</code>,
+        /// for OpenSSL 1.0.x this is the result of <code>SSLeay()</code>.
+        /// </remarks>
+        [UnsupportedOSPlatform("android")]
+        [UnsupportedOSPlatform("browser")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        [UnsupportedOSPlatform("windows")]
+        public static long OpenSslVersion
+        {
+            get
+            {
+                long version = Volatile.Read(ref _openSslVersion);
```

Comment:
What would happen if you don't use Volatile operations?

Reply:
Since the field is static, one thread could be reading the field and another could be writing to it. We don't want one thread reading a cached value if another thread updated it.

In theory if macOS supported 32-bit it could also result in torn reads, but since macOS doesn't, that's not a concern.

Reply:
> We don't want one thread reading a cached value if another thread updated it.

What would be a dangerous cached value? I'm think it would be if openSSL is updated in the middle on the process running, is that correct?

Reply:
Resolving since we don't need to use the volatile since its back in a cctor now.

Reply:
I guess I'm confused as well. A thread either sees 0 (and finds and writes the value) or the correct value. What does "reading while another is writing" mean if there is no tearing?